### ISSUE TITLE
When user set a proxy(Orbot), check its port first.

### ISSUE THREAD

**Issue by nonegit** [siacs/Conversations#2540]
https://github.com/siacs/Conversations/issues/2538

> What's up with that F-Droid issue?
Because fdroid have same problem.

> rather your USB reverse tether not properly creating network interfaces
> that make Android believe that there is a network connection

But I can use Firefox with this.
1. User enabled "Orbot" in your app's settings.
2. Your app try to connect:
a. Scan port 127.0.0.1:9050.
b. If a.IsPortOpen() === true, then connect. else, show error.

It's not about device. It's about trying portscan.

Next: how to set up reverse tether without root and app!

**Comment by nonegit** [comment]
Requirements:
a. Windows PC [or Linux if you know how to do it]
b. Smartphone (5.x or above)
c. Tor is installed on windows as a service or something.
d. Stone (download it to windows pc and android)

1. Connect to Windows PC with USB cable.
2. Smartphone: Turn OFF your wifi. Enable Airplane mode. Make sure it's offline completely.
3.  Go to settings - more - tethering.
4. Tap "USB tethering". Android will make a NIC on your windows.

5. Computer:
a. Open Control Panel\Network and Internet\Network Connections.
b. Select Android NIC and open its properties.
c. Disable IPv6, enable IPv4 only.
d. Set your IP address and mask to 192.168.42.50 / 255.255.255.0. (.50 <- just an example, change this to else)

6. Computer:
a. Open command prompt. And start stone.
stone.exe -d -n 127.0.0.1:9050 192.168.42.50:9050
(meaning: all connections go to .50:9050/tcp will redirect to localhost:9050/tcp)
b. Stone is open and running? Good. Back to smartphone.

7. Smartphone:
a. Install Stone for Android.
b. Open app. Input:
-d 192.168.42.50:9050/tcp 127.0.0.1:9050/tcp
(meaning: ...you know, okay?)
c. Tap RUN.
d. Running? Gooood!

8. Enjoy!
a. Open Firefox.
b. Set proxy to 127.0.0.1:9050, socks5, remoteDNS.
c. Enjoy internet via USB cable!!!!!!!!!!!!!!!!

Android(always 192.168.42.129) ===USB=== PC(192.168.42.50)---localhost---Tor===INTERNET

**Comment by nonegit** [comment]
LINKS

http://www.gcd.org/sengoku/stone/ <---- WINDOWS/LINUX STONE
https://apkpure.com/stone-for-android/jp.klab.stone <----- ANDROID STONE
https://www.torproject.org/download/download.html.en <----- TOR (to install as local service)

**Comment by nonegit** [comment]
https://gitlab.com/fdroid/fdroidclient/issues/1101

**Comment by nonegit** [comment]
TL;DR When the user enabled Orbot support, don't believe device status(such as Wifi offline/5G offline). Try a port scan against 127.0.0.1:9050 to decide online status.
